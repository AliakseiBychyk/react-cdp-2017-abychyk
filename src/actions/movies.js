export const RECEIVE_MOVIES = 'RECEIVE_MOVIES'

export const receiveMovies = (movies) => ({
  type: RECEIVE_MOVIES,
  movies  
})

export const fetchMoviesJson = (criterion, query) => { 
  const url = `https://api.themoviedb.org/3/search/${criterion}?api_key=${api_key}&query=${query}&include_adult=true`   

  return fetch(url, {
    method: 'GET'
  }).then(resp => resp.json())
    .then(data => {
      return criterion === 'movie'
        ? data.results
        : data.results[0].known_for
    })
    .catch(console.error)
}