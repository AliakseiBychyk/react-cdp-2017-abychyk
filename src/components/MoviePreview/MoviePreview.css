.moviePreview {
  text-align: left;
  float: left;
  display: block;
  padding: 10px;
  position: relative;
  width: 200px;
  height: 300px;
  margin: 8px;
  cursor: pointer;
}

.img {
  position: absolute;
  width: 200px;
  object-fit: cover;
}

.error {
  margin-top: 50px;
  color: red;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  font-weight: bold;
  font-size: 25px;
}