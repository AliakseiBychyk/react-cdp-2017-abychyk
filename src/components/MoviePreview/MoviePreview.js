import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styles from './MoviePreview.css'

class MoviePreview extends Component {
  handleClick = () => {
    console.log(this.props.show_title)
    this.props.onClick(this.props.id)
  }

  render() {
    if (this.props.errorcode) {
      return (
        <div className={styles.error}>
          <div>Error: {this.props.errorcode}</div>  
          {this.props.message}
          <div>Try another search.</div>
        </div>
      )
    }
    return (
      <div
        className={styles.moviePreview}
        onClick={this.handleClick}
      >
        <img
          src={this.props.poster}
          alt={this.props.show_title}
          className={styles.img} />
      </div>
    )
  }
}

MoviePreview.propTypes = {
  id: PropTypes.number,
  unit: PropTypes.number,
  show_title: PropTypes.string,
  director: PropTypes.string,
  poster: PropTypes.string,
  errorcode: PropTypes.number,
  message: PropTypes.string,
  onClick: PropTypes.func
}

export default MoviePreview