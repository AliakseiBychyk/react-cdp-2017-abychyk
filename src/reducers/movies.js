import { RECEIVE_MOVIES } from '../actions/movies'

export default (state = [], action) => {
  switch (action.type) {
    
    case RECEIVE_MOVIES:
      return action.movies

    default:
      return state
  }
}