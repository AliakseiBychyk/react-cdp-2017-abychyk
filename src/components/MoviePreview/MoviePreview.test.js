import React from 'react'
import { shallow } from 'enzyme'
import MoviePreview from './MoviePreview'
import movies from '../../fixtures/testingMovieQuery.json'
import error from '../../fixtures/errorResponse.json'

test('should render MoviePreview correctly', () => {
  const wrapper = shallow(<MoviePreview />)
  expect(wrapper).toMatchSnapshot()
})

test('should render MoviePreview with movie', () => {
  const onClick = jest.fn()
  const wrapper = shallow(<MoviePreview onClick={onClick} { ...movies[1] }/>)
  expect(wrapper).toMatchSnapshot()
})

test('should render MoviePreview with errorcode', () => {
  const wrapper = shallow(<MoviePreview { ...error }/>)
  expect(wrapper).toMatchSnapshot()
})