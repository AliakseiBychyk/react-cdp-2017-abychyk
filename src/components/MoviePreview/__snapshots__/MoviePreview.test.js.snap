// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MoviePreview correctly 1`] = `
<div
  className="moviePreview"
>
  <Link
    replace={false}
    to="/movie/undefined"
  >
    <img
      className="img"
    />
  </Link>
</div>
`;

exports[`should render MoviePreview with errorcode 1`] = `
<div
  className="error"
>
  <div>
    Error: 
    400
  </div>
  Sorry! Your input length must be greater than 5
  <div>
    Try another search.
  </div>
</div>
`;

exports[`should render MoviePreview with movie 1`] = `
<div
  className="moviePreview"
>
  <Link
    replace={false}
    to="/movie/The Terminator"
  >
    <img
      alt="The Terminator"
      className="img"
      src="http://netflixroulette.net/api/posters/1032625.jpg"
    />
  </Link>
</div>
`;
